rcode-reader/src/main/java/eu/rssw/pct/RCodeInfo.java

```diff
@@ -0,0 +1,723 @@
+      out.printf("%n*********%nSIGNATURE%n*********%n");
+      printByteBuffer(out, header);
+    }
+
+    int preambleSize = readAsciiEncodedNumber(header, 0, 4);
+    int numElements = readAsciiEncodedNumber(header, 4, 4);
+    // Version of signature block : offset 8, 4 bytes
+    // Encoding : offset 12, null-terminated string
+
+    int pos = preambleSize;
+    for (int kk = 0; kk < numElements; kk++) {
+      String str = readNullTerminatedString(header, pos);
+      pos += str.length() + 1;
+
+      // Datasets and temp-tables not read for now
+      if (str.startsWith("DSET") || str.startsWith("TTAB")) {
+        continue;
+      }
+
+      // Will probably be skipped
+      // Function fn = parseProcSignature(str);
+      // if ((unit.mainProcedure == null) && (fn.type == FunctionType.MAIN)) {
+      // unit.mainProcedure = fn;
+      // } else {
+      // unit.funcs.add(fn);
+      // }
+    }
+  }
+
+  private final void processSegmentTable(InputStream input, PrintStream out) throws IOException, InvalidRCodeException {
+    byte[] header = new byte[segmentTableSize];
+    int bytesRead = input.read(header);
+    if (bytesRead != segmentTableSize) {
+      throw new InvalidRCodeException("Not enough bytes in segment table block");
+    }
+    if (out != null) {
+      out.printf("%n*******%nSEGMENT%n*******%n");
+      printByteBuffer(out, header);
+    }
+
+    initialValueSegmentOffset = ByteBuffer.wrap(header, SEGMENT_TABLE_OFFSET_INITIAL_VALUE_SEGMENT_OFFSET, Integer.BYTES).order(order).getInt();
+    initialValueSegmentSize = ByteBuffer.wrap(header, SEGMENT_TABLE_OFFSET_INITIAL_VALUE_SEGMENT_SIZE, Integer.BYTES).order(order).getInt();
+    debugSegmentOffset = ByteBuffer.wrap(header, SEGMENT_TABLE_OFFSET_DEBUG_SEGMENT_OFFSET, Integer.BYTES).order(order).getInt();
+    debugSegmentSize = ByteBuffer.wrap(header, SEGMENT_TABLE_OFFSET_DEBUG_SEGMENT_SIZE, Integer.BYTES).order(order).getInt();
+  }
+
+  private final void processTypeBlock(InputStream input, PrintStream out) throws IOException, InvalidRCodeException {
+    byte[] segment = new byte[typeBlockSize];
+    int bytesRead = input.read(segment);
+    if (bytesRead != typeBlockSize) {
+      throw new InvalidRCodeException("Not enough bytes in type block");
+    }
+    if (out != null) {
+      out.printf("%n**********%nTYPE BLOCK%n***********%n");
+      printByteBuffer(out, segment);
+    }
+
+    int publicElementCount = ByteBuffer.wrap(segment, 8, Short.BYTES).order(order).getShort();
+    int protectedElementCount = ByteBuffer.wrap(segment, 10, Short.BYTES).order(order).getShort();
+    int privateElementCount = ByteBuffer.wrap(segment, 12, Short.BYTES).order(order).getShort();
+    int constructorCount = ByteBuffer.wrap(segment, 14, Short.BYTES).order(order).getShort();
+    int interfaceCount = ByteBuffer.wrap(segment, 16, Short.BYTES).order(order).getShort();
+    // int textAreaSize = ByteBuffer.wrap(segment, 24, Integer.BYTES).order(order).getInt();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3ACommentedOutCodeLine)